symengine/pow.h

```diff
@@ -34,12 +34,12 @@ class Pow : public Basic
     //! \return `true` if canonical
     bool is_canonical(const Basic &base, const Basic &exp) const;
     //! \return `base` of `base**exp`
-    inline RCP<const Basic> get_base() const
+    inline const RCP<const Basic>& get_base() const
     {
         return base_;
     }
     //! \return `exp` of `base**exp`
-    inline RCP<const Basic> get_exp() const
+    inline const RCP<const Basic>& get_exp() const
```

Comment:
> You only return RCP if you want to assign somewhere else (and increment a refcount), which you do at several places

I agree that you should return a RCP if you want to assign to somewhere else, but since `exp` is stored as a `RCP`, this is returning a reference to it.

Most of the classes have data members public and we can make them private by adding getters that return a const reference. In the case of RCP member, this is a `const RCP<const Basic>&`. What do you think?